lib/ansible/module_utils/crypto.py

```diff
@@ -128,6 +130,37 @@ def parse_name_field(input_dict):
     return result
 
 
+def convert_relative_to_datetime(relative_time_string):
+    """Get a datetime.datetime or None from a string in the time format described in sshd_config(5)"""
+
+    parsed_result = re.match(
+        r"^(?P<prefix>[+-])((?P<weeks>\d+)[wW])?((?P<days>\d+)[dD])?((?P<hours>\d+)[hH])?((?P<minutes>\d+)[mM])?((?P<seconds>\d+)[sS]?)?$",
```

Comment:
Months and years are NOT convertible to seconds, since their length varies by a day or more.

Reply:
Yep, but we're eventually working with dates here, not with date intervals.

Reply:
If there's a relative time, then that's an interval (e.g. "1 week from today") instead of a date ("2019-01-12"). What date is one month after January 30th?

Reply:
What I mean is that we're working with intervals, but from a well-defined absolute point (namely now). About the variable number of days in a month: that's why people are "rounding down" to the last day of the month. While that's not totally correct / well-defined, it usually does what people expect when they want to have a date 6 months in the future, or 10 years in the future.

Reply:
>  it usually does what people expect when they want to have a date 6 months in the future, or 10 years in the future

What date would you expect the function to give you for 10 years after 2012-02-29?

What date would you expect 6 months after August 31st in a leap year and a non-leap year?

Reply:
2022-02-28, and xxxx-02-28 for non-leap years and xxxx-02-29 for leap years. End of the month gets mapped to end of month. More interesting will be what I expect August 30th to be in 6 months :-)

Reply:
I'd rather have a bijective funtion (https://en.wikipedia.org/wiki/Bijection) than a surjective one (https://en.wikipedia.org/wiki/Surjective_function) because that way it leaves it up to the user to decide which edge case handling they prefer (at which point is the "end of month" mapping overrules the "only the month changes, not the day" rule).

I suspect this is the same reason why openssh also doesn't allow setting anything beyond weeks and why openssl only uses days as relative input.

Reply:
It's not even surjective; when your current date is Feburary 28, you can't reach Feburary 29 in the year after it (assuming it's a leap year).

It might be better to solve this once and for all as a lookup plugin which allows date arithmetic (currently you can use the `pipe` lookup to call `date`; see [here](https://groups.google.com/forum/#!msg/ansible-project/AX4BNH3eIfw/jrfISBDjDAAJ)) (date seems to keep days: `date --iso-8601 --date="2019-01-31 + 1 month"` yields `2019-03-03`).